crates/store/re_dataframe/src/range.rs

```diff
@@ -0,0 +1,265 @@
+            re_tracing::profile_scope!("queries");
+
+            columns
+                .iter()
+                .filter_map(|descr| match descr {
+                    ColumnDescriptor::Component(descr) => Some(descr),
+                    _ => None,
+                })
+                .filter_map(|descr| {
+                    let arrays = pov_time_column
+                        .times()
+                        .map(|time| {
+                            let query = LatestAtQuery::new(self.query.timeline, time);
+
+                            let results = self.engine.cache.latest_at(
+                                self.engine.store,
+                                &query,
+                                &descr.entity_path,
+                                [descr.component_name],
+                            );
+
+                            results
+                                .components
+                                .get(&descr.component_name)
+                                .and_then(|unit| {
+                                    unit.component_batch_raw(&descr.component_name).clone()
+                                })
+                        })
+                        .collect_vec();
+                    let arrays = arrays
+                        .iter()
+                        .map(|array| array.as_ref().map(|array| &**array as &dyn ArrowArray))
+                        .collect_vec();
+
+                    let list_array =
+                        re_chunk::util::arrays_to_list_array(descr.datatype.clone(), &arrays);
+
+                    if cfg!(debug_assertions) {
+                        #[allow(clippy::unwrap_used)] // want to crash in dev
+                        Some((descr, list_array.unwrap()))
+                    } else {
+                        // NOTE: Technically cannot ever happen, but I'd rather that than an uwnrap.
+                        list_array.map(|list_array| (descr, list_array))
+                    }
+                })
+                .collect()
+        };
+
+        // NOTE: Keep in mind this must match the ordering specified by `Self::schema`.
+        let packed_arrays = {
+            re_tracing::profile_scope!("packing");
+
+            columns
+                .iter()
+                .map(|descr| match descr {
+                    ColumnDescriptor::Control(_descr) => pov_chunk.row_ids_array().to_boxed(),
+
+                    ColumnDescriptor::Time(descr) => {
+                        let time_column = pov_chunk.timelines().get(&descr.timeline).cloned();
+
+                        if cfg!(debug_assertions) {
+                            #[allow(clippy::unwrap_used)] // want to crash in dev
+                            time_column.unwrap().times_array().to_boxed()
+                        } else {
+                            // NOTE: Technically cannot ever happen, but I'd rather that than an uwnrap.
```

Comment:
I believe this happens if the POV component was never logged on the requested (secondary) timeline.